### PR TITLE
Win7 fixes and enhancements

### LINKED ISSUE #27: Error uploading guest additions - breaks building Win7
Trying to build a Windows 7 box I get the following error:

```bash
gliverma@green-lizard /storage/repos/mwrock_packer-templates $ packer build vbox-win7-ent.json 
virtualbox-iso output will be in this color.

==> virtualbox-iso: Downloading or copying Guest additions
    virtualbox-iso: Downloading or copying: file:///usr/share/virtualbox/VBoxGuestAdditions.iso
==> virtualbox-iso: Downloading or copying ISO
    virtualbox-iso: Downloading or copying: file:///data/Downloads/_ISOs/Windows/en_windows_7_enterprise_with_sp1_x64_dvd_620201.iso
==> virtualbox-iso: Creating floppy disk...
    virtualbox-iso: Copying: answer_files/win7/Autounattend.xml
    virtualbox-iso: Copying: answer_files/win7/postunattend.xml
    virtualbox-iso: Copying: scripts/boxstarter.ps1
    virtualbox-iso: Copying: scripts/package.ps1
    virtualbox-iso: Copying: scripts/Test-Command.ps1
==> virtualbox-iso: Creating virtual machine...
==> virtualbox-iso: Creating hard drive...
==> virtualbox-iso: Attaching floppy disk...
==> virtualbox-iso: Creating forwarded port mapping for SSH (host port 2476)
==> virtualbox-iso: Executing custom VBoxManage commands...
    virtualbox-iso: Executing: modifyvm packer-virtualbox-iso-1449544140 --natpf1 winrm,tcp,,55985,,5985
    virtualbox-iso: Executing: modifyvm packer-virtualbox-iso-1449544140 --memory 5120
    virtualbox-iso: Executing: modifyvm packer-virtualbox-iso-1449544140 --vram 36
    virtualbox-iso: Executing: modifyvm packer-virtualbox-iso-1449544140 --cpus 2
==> virtualbox-iso: Starting the virtual machine...
==> virtualbox-iso: Waiting 10s for boot...
==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Waiting for WinRM to become available...
==> virtualbox-iso: Connected to WinRM!
==> virtualbox-iso: Uploading VirtualBox version info (5.0.10)
==> virtualbox-iso: Uploading VirtualBox guest additions ISO...
==> virtualbox-iso: Unregistering and deleting virtual machine...
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Error uploading guest additions: Error uploading file to $env:TEMP\winrmcp-56664c8e-4bda-7dff-10fd-2b2f6d4c06ae.tmp: Couldn't create shell: http error: 500 - <s:Envelope xml:lang="en-US" xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:x="http://schemas.xmlsoap.org/ws/2004/09/transfer" xmlns:e="http://schemas.xmlsoap.org/ws/2004/08/eventing" xmlns:n="http://schemas.xmlsoap.org/ws/2004/09/enumeration" xmlns:w="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd" xmlns:p="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd"><s:Header><a:Action>http://schemas.dmtf.org/wbem/wsman/1/wsman/fault</a:Action><a:MessageID>uuid:2DC4F698-E9B3-4BA4-BD5C-45E97A60D300</a:MessageID><a:To>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:To><a:RelatesTo>uuid:4a00c3b5-18d3-461e-6bf1-f1ab9c2cb2e5</a:RelatesTo></s:Header><s:Body><s:Fault><s:Code><s:Value>s:Receiver</s:Value><s:Subcode><s:Value>w:InternalError</s:Value></s:Subcode></s:Code><s:Reason><s:Text xml:lang="en-US">Server execution failed </s:Text></s:Reason><s:Detail><f:WSManFault xmlns:f="http://schemas.microsoft.com/wbem/wsman/1/wsmanfault" Code="2148007941" Machine="127.0.0.1"><f:Message><f:ProviderFault provider="Shell cmd plugin" path="%systemroot%\system32\winrscmd.dll">Server execution failed </f:ProviderFault></f:Message></f:WSManFault></s:Detail></s:Fault></s:Body></s:Envelope>

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Error uploading guest additions: Error uploading file to $env:TEMP\winrmcp-56664c8e-4bda-7dff-10fd-2b2f6d4c06ae.tmp: Couldn't create shell: http error: 500 - <s:Envelope xml:lang="en-US" xmlns:s="http://www.w3.org/2003/05/soap-envelope" xmlns:a="http://schemas.xmlsoap.org/ws/2004/08/addressing" xmlns:x="http://schemas.xmlsoap.org/ws/2004/09/transfer" xmlns:e="http://schemas.xmlsoap.org/ws/2004/08/eventing" xmlns:n="http://schemas.xmlsoap.org/ws/2004/09/enumeration" xmlns:w="http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd" xmlns:p="http://schemas.microsoft.com/wbem/wsman/1/wsman.xsd"><s:Header><a:Action>http://schemas.dmtf.org/wbem/wsman/1/wsman/fault</a:Action><a:MessageID>uuid:2DC4F698-E9B3-4BA4-BD5C-45E97A60D300</a:MessageID><a:To>http://schemas.xmlsoap.org/ws/2004/08/addressing/role/anonymous</a:To><a:RelatesTo>uuid:4a00c3b5-18d3-461e-6bf1-f1ab9c2cb2e5</a:RelatesTo></s:Header><s:Body><s:Fault><s:Code><s:Value>s:Receiver</s:Value><s:Subcode><s:Value>w:InternalError</s:Value></s:Subcode></s:Code><s:Reason><s:Text xml:lang="en-US">Server execution failed </s:Text></s:Reason><s:Detail><f:WSManFault xmlns:f="http://schemas.microsoft.com/wbem/wsman/1/wsmanfault" Code="2148007941" Machine="127.0.0.1"><f:Message><f:ProviderFault provider="Shell cmd plugin" path="%systemroot%\system32\winrscmd.dll">Server execution failed </f:ProviderFault></f:Message></f:WSManFault></s:Detail></s:Fault></s:Body></s:Envelope>

==> Builds finished but no artifacts were created.
```

### PR DESCRIPTION
**Pull request by genebean**
Fixes building Windows 7, adds support for Windows 7 Enterprise.  I had to add back in a line you removed that copies over the postunattend.xml as nothing was doing that in the win7 template anymore. The powershell upgrade is due to a known bug in PS v2 causing the install script of Virtualbox's guest additions to fail. Virtual CloneDrive was added as part of the tools install since Windows 7 can't mount an ISO. The guest part of the template mentioned below is so that the version of Virtualbox can be determined. 
```json
"guest_additions_mode": "disable",
"virtualbox_version_file": "C:/Users/vagrant/.vbox_version"
```
That part of Packer's process does not happen until after the build is done but before the provisioners... that is also way the tools are installed with a provisioner. Lastly, the reboots in the provisioning phase are to resolve issues that were keeping the tools from installing correctly.

I am happy to adjust this if needed. It has only been tested on Win7 Enterprise. 

Oh, this also fixes #27.

### CODE REVIEW

**Comment by genebean** [comment]
This is the PS bug I mentioned: http://www.leeholmes.com/blog/2008/07/30/workaround-the-os-handles-position-is-not-what-filestream-expected/

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Vagrant.configure(2) do |config|
```
Comment (mwrock) [inline Vagrantfile]: not sure if adding this was intentional, but probably doesn't belong here.

**Comment by mwrock** [comment]
Ahh. It looks like you took a somewhat similar approach as I did when I changed the 2012R2 box to install the guest additions. Moving that bit into a packer provisioner. Thats why the unattend was no longer in package.ps1. I cut a bunch of stuff out of that and moved it to the provisioner.

I so rarely work with win7 these days that and have not built a win7 box since doing that. I'll eventually try to merge that in to the win7 template.

File: Vagrantfile (hunk 1 above)
Comment (genebean) [inline Vagrantfile]: It was intentional. It's for testing the results of the builds.

**Comment by genebean** [comment]
Ok. I'm going to add this, and maybe some of your other things, into [genebean/packer_templates for #1](https://github.com/genebean/packer-templates/issues/1) to make a more complete setup. Thanks for the head start!

**Pull request closed without merging**